toolchain/lex/tokenized_buffer.cpp

```diff
@@ -325,14 +346,18 @@ class [[clang::internal_linkage]] TokenizedBuffer::Lexer {
                     NoWhitespaceAfterCommentIntroducer);
     }
 
-    // Now just consume the text until a newline.
-    while (!source_text.empty() && source_text.front() != '\n') {
-      ++current_column_;
-      source_text = source_text.drop_front();
+    // Use the current line info to jump to the end of the line.
+    source_text =
+        source_text.drop_front(current_line_info_->length - current_column_);
+    // This may be the end of the file in which case we immediately return.
+    if (source_text.empty()) {
+      // Finished lexing.
+      return;
     }
```

Comment:
From the writeup, it seems like the main win here is using `memchr` to find the end of the line instead of this manual scan. How does this PR compare to just changing the `drop_front` loop here to instead use `memchr`?

Reply:
Most of the comment line improvements are similar. They're also so noisy its hard for me to tell if there is a small difference.

I think the performance difference between the two approaches is probably in the noise, or if anything favors just using `memchr` here. I have some follow-on optimizations that might end up making pre-splitting somewhat faster. Haven't yet explored what we could do if we don't pre-split.

The question I guess is really whether you prefer the simplifications from finding lines in advance? I don't know how best to estimate those.

I'm also a bit less certain of how just scanning here will play out long-term with CR and CR+LF... But maybe we can make either work. WDYT?

Reply:
Doing the scanning in advance seems simpler and cleaner to me; if we're not paying an important performance penalty for it (and it sounds like we're not) then I think that's the right choice.